### PR TITLE
Init genesis in gov xmod

### PR DESCRIPTION
**Pull request by NYBACHOK**
Part of https://github.com/rumos-io/gears/issues/117

### CODE REVIEW

File: x/gov/src/keeper.rs [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+           Okay. I think that in our implementation there is no need to create account if it.
+
+           So I should omit this lines...
+           if balance.is_empty() || balance.iter().any(|this| this.amount.is_zero()) {
+               https://github.com/cosmos/cosmos-sdk/blob/d3f09c222243bb3da3464969f0366330dcb977a8/x/gov/genesis.go#L47
+           }
+        */
+
+        if !(balance == total_deposits) {
+            panic!(
```
Comment (joneskm) [inline x/gov/src/keeper.rs:130]: So we're expecting that in the bank module init genesis the gov_mod account will be initialized to the right amount?

File: x/gov/src/types/deposit.rs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    pub depositor: AccAddress,
+    pub amount: Vec<Coin>,
+}
+
+impl Deposit {
+    pub(crate) fn key(&self) -> Vec<u8> {
+        [
+            KEY_DEPOSIT_PREFIX.as_slice(),
+            &self.proposal_id.to_be_bytes(),
+            &[self.depositor.len() as u8], // We save 'cause `AccAddress` len shoudn't be bigger than 255
```
Comment (joneskm) [inline x/gov/src/types/deposit.rs:18]: len() returns a u8, so we don't need the cast

File: x/gov/src/types/proposal.rs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct Proposal {
+    pub proposal_id: u64,
+    pub content: Vec<u8>, // TODO
+    pub status: ProposalStatus,
+    pub final_tally_result: (), // TODO: https://github.com/cosmos/cosmos-sdk/blob/d3f09c222243bb3da3464969f0366330dcb977a8/x/gov/types/gov.pb.go#L289
+    pub submit_time: (),
+    pub deposit_end_time: DateTime<Utc>,
+    pub total_deposit: Vec<Coin>,
```
Comment (joneskm) [inline x/gov/src/types/proposal.rs:21]: Are there any additional constraints on the list of coins? Do we need a new type?

File: x/gov/src/types/vote.rs [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    pub voter: AccAddress,
+    pub options: (), // TODO:
+}
+
+impl Vote {
+    pub fn key(&self) -> Vec<u8> {
+        [
+            KEY_VOTES_PREFIX.as_slice(),
+            &self.proposal_id.to_be_bytes(),
+            &[self.voter.len() as u8], // We save 'cause `AccAddress` len shoudn't be bigger than 255
```
Comment (joneskm) [inline x/gov/src/types/vote.rs:18]: Same as before. len() returns a u8.

File: x/gov/src/types/deposit.rs (hunk 2 above)
Comment (NYBACHOK) [reply x/gov/src/types/deposit.rs:18]: Unexpected for me...

File: x/gov/src/types/proposal.rs (hunk 3 above)
Comment (NYBACHOK) [reply x/gov/src/types/proposal.rs:21]: For now, it's okay, but later maybe I add a new type or use `SendCoins`( it needs to be renamed)

File: x/gov/src/keeper.rs (hunk 1 above)
Comment (NYBACHOK) [reply x/gov/src/keeper.rs:130]: Right. But I need to make sure that it created before usage(I suspected that It doesn't) and additional thing to note is denom.

File: x/gov/src/types/deposit.rs (hunk 2 above)
Comment (NYBACHOK) [reply x/gov/src/types/deposit.rs]: Removed

File: x/gov/src/types/vote.rs (hunk 4 above)
Comment (NYBACHOK) [reply x/gov/src/types/vote.rs]: Removed

File: x/gov/src/types/proposal.rs (hunk 3 above)
Comment (joneskm) [reply x/gov/src/types/proposal.rs:21]: Fair enough. I agree with the renaming, `Vec<Coin>` seems to be appearing more and more.

**Pull request merged**